src/coreclr/vm/callhelpers.cpp

```diff
@@ -317,7 +317,8 @@ void MethodDescCallSite::CallTargetWorker(const ARG_SLOT *pArguments, ARG_SLOT *
                  || isCallConv(m_methodSig.GetCallingConvention(), CorCallingConvention(IMAGE_CEE_CS_CALLCONV_C))
                  || isCallConv(m_methodSig.GetCallingConvention(), CorCallingConvention(IMAGE_CEE_CS_CALLCONV_VARARG))
                  || isCallConv(m_methodSig.GetCallingConvention(), CorCallingConvention(IMAGE_CEE_CS_CALLCONV_NATIVEVARARG))
-                 || isCallConv(m_methodSig.GetCallingConvention(), CorCallingConvention(IMAGE_CEE_CS_CALLCONV_STDCALL)));
+                 || isCallConv(m_methodSig.GetCallingConvention(), CorCallingConvention(IMAGE_CEE_CS_CALLCONV_STDCALL))
+                 || isCallConv(m_methodSig.GetCallingConvention(), CorCallingConvention(IMAGE_CEE_CS_CALLCONV_UNMANAGED)));
```

Comment:
This assert has no value. It would be better to delete it and change the ifdef to `#ifndef FEATURE_INTERPRETER`.

Reply:
The callconv of `IMAGE_CEE_CS_CALLCONV_UNMANAGED` comes from ILStub which is generated to call `Sys:LChflagsCanSetHiddenFlag` in libSystem.Native.so.

The code which sets callconv to `IMAGE_CEE_CS_CALLCONV_UNMANAGED` is found here:
https://github.com/dotnet/runtime/blob/c3940f603a42e5c78f888ba2cbab76fbfda29d83/src/coreclr/vm/stubgen.cpp#L2381-L2385

I have also noticed `AddCallConvModOpt` is called later:
https://github.com/dotnet/runtime/blob/f8629e62f7c8486af108a9ae10fec65e7c85c6af/src/coreclr/vm/stubgen.cpp#L2790-L2800

But I am still not sure how to add this case to those asserts when the callconv is `IMAGE_CEE_CS_CALLCONV_UNMANAGED` here.

Reply:
Right, it is why I have suggested to delete this assert.